packages/components/src/ErrorBoundary.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, PropsWithChildren } from 'react';
+
+export type ErrorBoundaryProps = PropsWithChildren<{
+  onError?: (error: unknown) => void;
+}>;
+
+type ErrorBoundaryState = {
+  hasError: boolean;
+};
+
+export class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
```

Comment:
interesting... is the guidance to still do ErrorBoundary implementations as a React class component versus a functional one?

Reply:
> There is currently no way to write an error boundary as a function component

https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary